mantidproject/mantid
# Refactor threading on sans gui

**Description of work.**
The SANS GUI would previously hang if it could not find the correct run files whilst it searched the archive. This updates the GUI so that file searching is safely done in a different thread. 
**Report to:** [user name]/[nobody]. 

**To test:**
[loqdemo.zip](https://github.com/mantidproject/mantid/files/2428250/loqdemo.zip)
* Get the data from the link above load MaskFile.txt as the user file and batch_mode_reduction.csv as the batch file.
* Check that the sample thickness column is updated to be 1.0
* Change the value in the Sample Scatter column to 740440, wait a bit but check that the GUI remains responsive whilst the background task runs. Confirm that eventually the row goes blue and that when you hover over it a tooltip displaying a useful error message is displayed. 
* Try playing around with other row edits and changes and check that the GUI remains responsive.




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
